### PR TITLE
 BaSP-MR-09-19: Update activity

### PR DESCRIPTION
**Pull request by octavi01**
I worked on the PUT req. Update an activity.

### CODE REVIEW

File: src/controllers/activity.js [hunk 1]
```diff
@@ -33,8 +33,37 @@ const getActivityById = (req, res) => {
       });
     });
 };
+// const Activity = require('../models/Activity'); that was mi code before the PULL,with "A"ctivity
```
Comment (NicoCagnina) [inline src/controllers/activity.js:36]: remove this comment

File: src/routes/activity.js [hunk 2]
```diff
@@ -1,10 +1,14 @@
 const express = require('express');
+const activityController = require('../controllers/activity');
+const validations = require('../validations/activity');
 
 const controller = require('../controllers/activity');
 
 const router = express.Router();
 
 router
+  // .delete('/:id', activityController.deleteActivity)
```
Comment (NicoCagnina) [inline src/routes/activity.js:10]: remove this comment

**Review by NicoCagnina — CHANGES_REQUESTED** [review]

File: src/controllers/activity.js [hunk 3]
```diff
@@ -34,7 +34,35 @@ const getActivityById = (req, res) => {
...
+    { new: true },
+  )
+    .then((result) => {
+      if (!result) {
+        return res.status(404).json({
+          message: `Activity with the id: ${id} was not found, please try with another one`,
+          error: true,
+        });
+      }
+      return res.status(201).json(result);
```
Comment (JuanMornik) [inline src/controllers/activity.js:59]: please add a message

File: src/controllers/activity.js [hunk 4]
```diff
@@ -34,7 +34,35 @@ const getActivityById = (req, res) => {
...
+    })
+    .catch((error) => res.status(400).json(error));
```
Comment (JuanMornik) [inline src/controllers/activity.js:61]: add a message here too

File: src/validations/activity.js [hunk 5]
```diff
@@ -0,0 +1,23 @@
+const Joi = require('joi');
+
+const validateUpdate = (req, res, next) => {
+  const activityValidation = Joi.object({
+    name: Joi.string().pattern(/^[A-Za-z\s]+$/).min(2).messages({
+      'string.pattern.base': 'The name must contain only letters',
+    }),
+    description: Joi.string().regex(/^[a-zA-Z0-9]{5,}$/),
```
Comment (JuanMornik) [inline src/validations/activity.js:8]: validate min and max length in this string

**Review by JuanMornik — CHANGES_REQUESTED** [review]

File: src/controllers/activity.js [hunk 6]
```diff
@@ -34,7 +34,41 @@ const getActivityById = (req, res) => {
...
+          message: `Activity with the id: ${id} was not found, please try with another one`,
+          error: true,
+        });
+      }
+      return res.status(201).json({
+        message: 'Activity updated successfully',
+        result,
+      });
+    })
+    .catch((error) => res.status(400).json({
```
Comment (NicoCagnina) [inline src/controllers/activity.js:64]: this should be 500

File: src/models/Activity.js [hunk 7]
```diff
@@ -7,14 +7,15 @@ const activitySchema = new Schema({
...
-    type: Boolean,
-    default: false,
-  },
   description: {
     type: String,
     required: true,
   },
+  isActive: {
+    type: Boolean,
+    required: true,
```
Comment (NicoCagnina) [inline src/models/Activity.js:16]: if we have a default we don't need a required

**Review by NicoCagnina — CHANGES_REQUESTED** [review]

**Review by NicoCagnina — APPROVED** [review]

**Review by JuanMornik — APPROVED** [review]

**Pull request merged**